lib/ddtrace/sampler.rb

```diff
@@ -6,22 +6,44 @@
 require 'ddtrace/diagnostics/health'
 
 module Datadog
-  # \Sampler performs client-side trace sampling.
+  # Interface for client-side trace sampling.
+  # @abstract
   class Sampler
+    # Returns `true` if the provided trace should be kept.
+    # Otherwise, `false`.
+    #
+    # This method *must not* modify the `trace`.
+    #
+    # @param [Datadog::Trace] trace
+    # @return [Boolean] should this trace be kept?
     def sample?(_trace)
       raise NotImplementedError, 'Samplers must implement the #sample? method'
     end
 
+    # Returns `true` if the provided trace should be kept.
+    # Otherwise, `false`.
+    #
+    # This method *may* modify the `trace`, in case changes are necessary based on the
+    # sampling decision.
+    #
+    # @param [Datadog::Trace] trace
+    # @return [Boolean] should this trace be kept?
     def sample!(_trace)
       raise NotImplementedError, 'Samplers must implement the #sample! method'
     end
 
+    # The sampling rate, if this sampler has such concept.
+    # Otherwise, `nil`.
+    #
+    # @param [Datadog::Trace] trace
+    # @return [Float,nil] sampling ratio between 0.0 and 1.0 (inclusive), or `nil` if not applicable
     def sample_rate(_trace)
       raise NotImplementedError, 'Samplers must implement the #sample_rate method'
     end
   end
 
   # \AllSampler samples all the traces.
+  # @public_api
   class AllSampler < Sampler
```

Comment:
Is it just me that finds it weird that AllSampler is at the top level at `Datadog::AllSampler`? Seems kinda a prominent location for such trivial details.

Reply:
It's kind of weird.